### PR TITLE
fix: update android sdk to range

### PR DESCRIPTION
**Pull request by mrehan27**
Closes: https://github.com/customerio/issues/issues/9429

### Changes

- Updated Android SDK to range so customer can get newer minor versions by just rebuilding the project

### CODE REVIEW

File: android/gradle.properties [hunk 1]
```diff
@@ -1,4 +1,4 @@
 customerio.reactnative.kotlinVersion=1.5.31
 customerio.reactnative.compileSdkVersion=30
 customerio.reactnative.targetSdkVersion=30
-customerio.reactnative.cioSDKVersionAndroid=3.3.0-beta.1
+customerio.reactnative.cioSDKVersionAndroid=3.3.+
```
Comment (Shahroz16) [inline android/gradle.properties:4]: We are allowing minor updates as well. 

`Major/Breaking.Minor.Fix/Patch`

```suggestion
customerio.reactnative.cioSDKVersionAndroid=[3.3,4.0)
```

**Review by Shahroz16 — APPROVED** [review]

**Review by levibostian — DISMISSED** [review]
Going to reject this PR to bring attention to [my rejected comment in this PR](https://github.com/customerio/customerio-flutter/pull/24)

**Comment by Shahroz16** [comment]
Update on [Levis's comment. ](https://github.com/customerio/customerio-flutter/pull/24#issuecomment-1433563434)

**Comment by mrehan27** [comment]
I tried listing dependencies locally, and `app:androidDependencies` task outputs following versions:

```
+--- io.customer.android:messaging-push-fcm:3.3.0-beta.1@aar
+--- io.customer.android:messaging-in-app:3.3.0-beta.1@aar
+--- io.customer.android:tracking:3.3.0-beta.1@aar
+--- build:gist:3.0.3@aar
```

So the range looks fine to me.

**Review by Shahroz16 — APPROVED** [review]

**Pull request merged**